src/main/java/com/criteo/nosql/cassandra/exporter/JmxScraper.java

```diff
@@ -312,10 +312,8 @@ private void updateMetric(MBeanServerConnection beanConn, MBeanInfo mBeanInfo, N
                 //Most beans declared as Object are Double in disguise
                 if (first >= '0' && first <= '9') {
                     updateStats(nodeInfo, mBeanInfo.metricName, Double.valueOf(str));
-
-                    // https://books.google.fr/books?id=BvsVuph6ehMC&pg=PA82
-                    // EstimatedHistogram are object for JMX but are long[] behind
-                } else if (first == '{' || mBeanInfo.metricName == "org:apache:cassandra:metrics:compaction:pendingtasksbytablename:value") {
+                    
+                } else if (first == '{' && mBeanInfo.metricName.equalsIgnoreCase("org:apache:cassandra:metrics:compaction:pendingtasksbytablename:value")) {
                     HashMap<String, HashMap<String, Integer>> pendingTasks = (HashMap<String, HashMap<String, Integer>>) value;
```

Comment:
If it works, put Int instead of Integer, as Int is a primitive type while Integer is a full-blown object

Reply:
It doesn't. You have to use integer